lib/vega_lite/data.ex

```diff
@@ -327,6 +327,328 @@ defmodule VegaLite.Data do
+    * `:grid_color` - backround color for the grid. Defaults to `"white"`.
+    * `:grid_stroke_color` - stroke color for the grid markings. Defaults to `"black"`.
+    * `:grid_stroke_opacity` - opacity for the grid strokes. Defaults to `1`.
+    * `:scheme` - the color scheme for multi-line plots. Defaults to `"turbo"`
+
+  ## Examples
+
+  In the example below, we plot a dataset by specifying stylized point marks and connecting it with
+  an interpolated line.
+
+      data = %{"r" => [1, 2, 3, 3, 4], "theta" => [0, 30, 45, 135, 270]}
+
+      VegaLite.Data.polar_plot(
+        VegaLite.new(title: "Polar Plot", height: 500, width: 500),
+        [
+          {data, :point, [stroke: "white", stroke_width: 3, tooltip: [data: true]]},
+          {data, :line, point: true, interpolate: "cardinal", color: "black", tooltip: [data: true]}
+        ],
+        angle_marks: [0, 15, 30, 45, 60, 90, 180, 270, 360],
+        radius_marks: [0, 1, 2, 3, 4, 5]
+      )
+
+  In the next example, we add a second line to the plot grouping by the `:legend_key`.
+  Note that the `:legend_key` will also be the legend title, so a readable name is desirable.
+
+      legend_key = "Line Groups"
+
+      data = %{
+        "r" => [1, 2, 3, 3, 4],
+        "theta" => [0, 30, 45, 135, 270],
+        legend_key => List.duplicate("First Line", 5)
+      }
+
+      other_data = %{
+        "r" => [8, 6, 4, 2, 0],
+        "theta" => [0, 30, 60, 90, 120],
+        legend_key => List.duplicate("Second Line", 5)
+      }
+
+      VegaLite.Data.polar_plot(
+        VegaLite.new(title: "Polar Plot", height: 500, width: 500),
+        [
+          {data, :point, [stroke: "black", stroke_width: 3]},
+          {data, :line, point: true, interpolate: "cardinal"},
+          {other_data, :point, [stroke: "black", stroke_width: 3]},
+          {other_data, :line, point: true, interpolate: "cardinal"}
+        ],
+        angle_marks: [0, 15, 30, 45, 60, 90, 180, 270, 360],
+        legend_key: legend_key
+      )
+  """
+  def polar_plot(vl \\ Vl.new(), data, opts \\ []) do
+    opts =
+      Keyword.validate!(
+        opts,
+        [
+          :radius_marks,
+          :legend_key,
+          direction: :counter_clockwise,
+          angle_marks: [0, 90, 180, 270, 360],
+          theta_offset: 0,
+          grid_opacity: 1,
+          grid_color: "white",
+          grid_stroke_color: "black",
+          grid_stroke_opacity: 1,
+          scheme: "turbo"
+        ]
+      )
+
+    data =
+      case data do
+        {%{} = data, mark} ->
+          [{data, mark, []}]
+
+        [_ | _] = data ->
+          Enum.map(data, fn
+            {data, mark} -> {Map.new(data, fn {k, v} -> {to_string(k), v} end), mark, []}
+            {data, mark, opts} -> {Map.new(data, fn {k, v} -> {to_string(k), v} end), mark, opts}
+          end)
+      end
+
+    opts =
+      case opts[:radius_marks] do
+        nil ->
+          radius_marks = polar_infer_radius_marks(data)
```

Comment:
One alternative is to infer the min and max using VegaLite, if at all possible.

Reply:
I wouldn't be because we plot separate layers for each radius mark

Reply:
Keep in mind though you can keep the data on the root, outside of the layers, then all layers will have access to it. You can see how other functions do it. My suggestion would be to trim this down into the smallest possible function, that follows the other patterns in this module, and then you can add all of the conveniences later in a pull request.

Reply:
I was wrong about them being separate layers, so maybe we can do some kind of vega-lite calculation for this. Though I don't see this inference missing as a blocker for the feature

Reply:
Keeping the data in the root hurts composability in a sense because I have different r/theta definitions depending on whether we're dealing with the angle marks, the radius marks or the data layers